### DISCUSSION TITLE
How to identify a specific STREAM on the peer side?

### DISCUSSION THREAD

**Discussion by chemsallioua** [microsoft/msquic#4258] [Q&A] [answered]
I am making a Server and a Client based on MSQUIC. The Client opens a connection, the Server opens 6 unidirectional STREAMs and one bidirectional STREAM. 

Now on the Client side, how can I identify which STREAM is the bidirectional one? What about identifying each of them?
Can I set a custom ID for the streams?

Thank you!

**Comment by nibanks** [comment]
The `PEER_STREAM_STARTED` event has the following:
```c
QUIC_STREAM_OPEN_FLAGS Flags;
```
This tells you if it's bidirectional or unidirectional:
```c
typedef enum QUIC_STREAM_OPEN_FLAGS {
    QUIC_STREAM_OPEN_FLAG_NONE              = 0x0000,
    QUIC_STREAM_OPEN_FLAG_UNIDIRECTIONAL    = 0x0001,   // Indicates the stream is unidirectional.
    QUIC_STREAM_OPEN_FLAG_0_RTT             = 0x0002,   // The stream was opened via a 0-RTT packet.
    QUIC_STREAM_OPEN_FLAG_DELAY_ID_FC_UPDATES = 0x0004, // Indicates stream ID flow control limit updates for the
                                                        // connection should be delayed to StreamClose.
} QUIC_STREAM_OPEN_FLAGS;
```

**Reply by chemsallioua** [reply]
Other than the bidirectional or not, how can I map them all individually from Server to Client?

**Reply by nibanks** [reply]
You can query the QUIC stream ID via GetParam.

**Reply by chemsallioua** [reply]
Yes, but the ID gets generated by the library, I do not have control over it. Right?

**Reply by nibanks** [reply, marked as answer]
Correct. If you want something your own, then you'll need to send it as payload.

**Reply by chemsallioua** [reply]
Okay clear. Thank you!
